tangelo/algorithms/variational/vqe_solver.py

```diff
@@ -125,6 +126,9 @@ def __init__(self, opt_dict):
                 warnings.warn("Spin-orbital ordering shifted to all spin-up first then down to ensure efficient generator screening "
                               "for the Jordan-Wigner mapping with QCC-based ansatze.", RuntimeWarning)
                 self.up_then_down = True
+            if self.ansatz == BuiltInAnsatze.pUCCD and self.qubit_mapping.lower() != "hcb":
+                warnings.warn("Forcing the hard-core boson mapping for the pUCCD ansatz.", RuntimeWarning)
```

Comment:
Is that the official terminology ? Genuinely curious :)

Reply:
Yep, at least as I saw in the paper. It refers to boson that cannot occupy the same energy levels (so obeying fermionic statistics). The advantage is that we do not have to consider anticommutation relations, i.e. long strings of Z Pauli matrices in the JW mapping.

It is similar to phaseless fermions; the distinction is that we have half the number of particle to consider (2 fermions = 1 boson in this approximation).